velox/exec/tests/RowContainerTest.cpp

```diff
@@ -363,17 +522,130 @@ class RowContainerTest : public exec::test::RowContainerTestBase {
             return rowContainer->compareRows(
                        l.second, r.second, {{nullsFirst, ascending}}) < 0;
           });
-      std::vector<std::optional<T>> sorted;
-      for (const auto& irow : indexedRows) {
-        if (decoded.isNullAt(irow.first)) {
-          sorted.push_back(std::nullopt);
-        } else {
-          sorted.push_back(decoded.valueAt<T>(irow.first));
-        }
+      auto result =
+          createResultVectorForArrayComparison<T>(indexedRows, decoded);
+      assertEqualVectors(expected, result);
+    }
+
+    // Verify compareRows method with default compare flags.
+    if (ascending && nullsFirst) {
+      std::sort(
+          indexedRows.begin(),
+          indexedRows.end(),
+          [&](const std::pair<int, char*>& l, const std::pair<int, char*>& r) {
+            return rowContainer->compareRows(l.second, r.second) < 0;
+          });
+      auto result =
+          createResultVectorForArrayComparison<T>(indexedRows, decoded);
+      assertEqualVectors(expected, result);
+    }
+  }
+
+  template <typename T>
+  VectorPtr createResultVectorForMapComparison(
```

Comment:
This template method, createResultVectorForArrayComparison and createResultVectorForSimpleComparison can be replaced with a simpler non-template method:

```
  VectorPtr copyValues(
      const IndexedRows& indexedRows,
      const VectorPtr& values) {
    const auto numRows = indexedRows.size();
    auto copy = BaseVector::create(values->type(), numRows, pool());
    for (auto i = 0; i < numRows; ++i) {
      copy->copy(values.get(), i, indexedRows[i].first, 1);
    }
    return copy;
  }

      auto result = copyValues(indexedRows, values);
      assertEqualVectors(expected, result);
`